service.rb

```diff
@@ -43,10 +43,22 @@ def initialize(port)
           line = client.readline
           if line == "\r\n"
             body = client.read(content_length)
+            break
           end
         end
 
-        response_message =  "WIP: POST RESPONSE"
+        new_daily_data = Hash[URI.decode_www_form(body)]
+
+        all_data << new_daily_data.transform_keys(&:to_sym)
+
+        # Copy paste from GET /show/data
+        response_message = ""
+        response_message << "<ul>\n"
+        all_data.each do |daily_data|
+          response_message << "<li> On this day <b>#{CGI.escapeHTML(daily_data[:date])}</b>, #{CGI.escapeHTML(daily_data[:step_count])}, #{CGI.escapeHTML(daily_data[:notes])}</li>\n"
+        end
+        response_message << "</ul>\n"
+        response_message << daily_data_form
```

Comment:
You‘re choosing to reproduce the content of the main page here rather than doing something else, like showing a success message with a link back to the main page, or using a different HTTP status code (like [201](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201) or [301](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301)) to send the browser back there automatically. That clearly works (copy-pasting aside), but e.g. what happens when the user reloads the page after submitting the form?

Reply:
Ah with this implementation and if the user reloads, the form is resubmitted for a second time. I modified the implementation: 
* Swap the response status code to 201 Created
* Add a location response header that redirects the page back to `/show/data`
* Remove the copy pasta code :)

Reply:
👏🏻

Reply:
(This illustrates why it doesn’t matter what path you `POST` to — the user doesn’t actually end up there after the redirect.)